spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkBucketFunction.java

```diff
@@ -351,6 +350,68 @@ public void testThatMagicFunctionsAreInvoked() {
             "staticinvoke(class org.apache.iceberg.spark.functions.BucketFunction$BucketBinary");
   }
 
+  @Test
+  public void testAnalyzedToApplyFunctionExpression() {
+    Assumptions.assumeThat(systemFunctionPushDownEnabled).isTrue();
+    // TinyInt
+    Assertions.assertThat(scalarSql("EXPLAIN EXTENDED SELECT system.bucket(5, 6Y)"))
+        .asString()
+        .isNotNull()
+        .contains("applyfunctionexpression(Wrapper(iceberg.bucket(int))");
```

Comment:
I don't think these tests are needed if we don't trick Spark into not using `StaticInvoke`.